cubequery/tasks/__init__.py

```diff
@@ -166,6 +180,105 @@ def ping_results(self, output_url, results):
+                                        if not (int(args[c]) >= condition['value'][0]) or not(int(args[c]) <= condition['value'][1]):
+                                            errors.append(create_error_message(condition))
+                                    else:
+                                        if not (int(args[c]) >= condition['value'][0]):
+                                            errors.append(create_error_message(condition))
+
+                        # Date Range Validation
+                        if condition['type'] == 'date_range':
+                            for c in condition['id']:
+                                if c in args:
+                                    if len(condition['value'])==2:
+                                        if not (args[c] >= condition['value'][0]) or not(args[c] <= condition['value'][1]):
+                                            errors.append(create_error_message(condition))
+                                    else:
+                                        if not (args[c] >= condition['value'][0]):
+                                            errors.append(create_error_message(condition))
+        
+        return errors
+
+    # TODO: Bounds conversion and sometimes spatial query dependent on product
+    def validate_standard_spatial_query(self, value):
+
+        errors = []
+
+        try:
+            parsed_polygon = wkt.loads(value)
+        except:
+            return [create_error_message({'id':'aoi', 'error_message':'Polygon could not be loaded', '_comment':'Polygon could not be loaded'})]
+
+        
+        '''
+        Returns validity of geometery (bool)
+        * Whole of Fiji = True
+        * Suva = True
+        '''
+        valid_geom = parsed_polygon.is_valid
+        if not valid_geom:
+            errors.append(create_error_message({'id':'aoi', 'error_message':'Geometry not a valid polygon', '_comment':'Geometry not a valid polygon'}))
+
+        '''
+        Returns area of polygon - About 1/4 of country ... 0.25 
+        * Whole of Fiji = 1.8662849915034905
+        * Suva = 0.017204474747948426
+        '''
+        area = parsed_polygon.area
+        if area > 0.25:
+            errors.append(create_error_message({'id':'aoi', 'error_message':'AOI area is too large', '_comment':'Size of polygon is too large'}))
+
+        '''
+        Returns bool for polygon inside Fiji
+        
+        For when antimeridian problem is resolved:
+        with open("TM_FIJI_BORDERS.geojson") as f:
+            features = json.load(f)["features"]
+            fiji_polygon = GeometryCollection([shape(feature["geometry"]).buffer(0) for feature in features])
+        '''
+            
+        fiji_polygon = wkt.loads('POLYGON((179.85466581578783 -15.573479020580137,177.13005644078854 -15.573479020580137,177.13005644078854 -19.40771171412428,179.85466581578783 -19.40771171412428,179.85466581578783 -15.573479020580137))')

```

Comment:
this wkt string needs to come from the config system. Cube query cant be tied to Fiji. It might be used else where in the world and this line locks it so it can only work in Fiji with out code changes.